pkg/sql/catalog/descs/descriptor.go

```diff
@@ -339,8 +339,10 @@ func getDescriptorsByID(
 	if err := tc.finalizeDescriptors(ctx, txn, flags, descs, vls); err != nil {
 		return err
 	}
-	if err := tc.hydrateDescriptors(ctx, txn, flags, descs); err != nil {
-		return err
+	if !flags.SkipHydration {
+		if err := tc.hydrateDescriptors(ctx, txn, flags, descs); err != nil {
```

Comment:
hydrateDescriptors also gets called by `GetLeasedImmutableTableByID` so perhaps the `flags.SkipHydration` check should happen within `hydrateDescriptors` itself?

Reply:
we may do that indeed.